File: 0896-monotonic-array/0896-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        if len(nums) == 1: return True
        increasing = (nums[0] < nums[-1])
        for i in range(0,len(nums) - 1):
            if (increasing and nums[i+1] < nums[i]) or (not increasing and nums[i+1] > nums[i]): 
                return False
        return True

File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int vectorSize = nums.size();
        bool increasing = (nums[0] < nums.back());
        for (int i = 0; i < nums.size() - 1; i++)
        {
            if (increasing){
                if(nums[i + 1] < nums[i]){return false;}
            }
            else{
                if (nums[i + 1] > nums[i]{return false}
            }

        }
        return true;
    }
};